// @mui-treasury/mockup/brands/dailyShopping/dailyShoppingTheme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _createMuiTheme = _interopRequireDefault(require("@material-ui/core/styles/createMuiTheme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = (0, _createMuiTheme["default"])({
  typography: {
    fontFamily: '"Fira Sans", "Roboto", "Helvetica", "Arial", sans-serif',
    h1: {
      fontFamily: '"Playfair Display", "Roboto", "Helvetica", "Arial", sans-serif'
    },
    h2: {
      fontFamily: '"Playfair Display", "Roboto", "Helvetica", "Arial", sans-serif'
    },
    h3: {
      fontFamily: '"Playfair Display", "Roboto", "Helvetica", "Arial", sans-serif'
    },
    h4: {
      fontFamily: '"Playfair Display", "Roboto", "Helvetica", "Arial", sans-serif'
    },
    h5: {
      fontFamily: '"Playfair Display", "Roboto", "Helvetica", "Arial", sans-serif'
    },
    h6: {
      fontFamily: '"Playfair Display", "Roboto", "Helvetica", "Arial", sans-serif'
    }
  },
  palette: {
    primary: {
      main: '#3E89EB'
    },
    background: {
      "default": '#ffffff'
    }
  }
});

exports["default"] = _default;
