// @mui-treasury/mockup/brands/dailyShopping/DailyHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Box = _interopRequireDefault(require("@material-ui/core/Box"));

var _Toolbar = _interopRequireDefault(require("@material-ui/core/Toolbar"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _ClearAll = _interopRequireDefault(require("@material-ui/icons/ClearAll"));

var _ShoppingBasket = _interopRequireDefault(require("@material-ui/icons/ShoppingBasket"));

var _DailyLogo = _interopRequireDefault(require("./DailyLogo"));

var _DailySearch = _interopRequireDefault(require("./DailySearch"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var useStyles = (0, _styles.makeStyles)(function () {
  return {
    cart: {
      backgroundColor: '#F6F7F9',
      borderRadius: 40,
      minWidth: 24,
      minHeight: 24,
      lineHeight: '24px',
      textAlign: 'center',
      padding: '0 8px',
      marginLeft: 8
    },
    flexCenter: {
      display: 'flex',
      alignItems: 'center'
    }
  };
});

var DailyHeader = function DailyHeader() {
  var styles = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_Toolbar["default"], {
    disableGutters: true
  }, /*#__PURE__*/_react["default"].createElement(_DailyLogo["default"], null), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    ml: 'auto'
  }, /*#__PURE__*/_react["default"].createElement(_DailySearch["default"], null)), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    ml: {
      xs: 2,
      md: 10,
      lg: 15
    },
    className: styles.flexCenter,
    justifyContent: 'flex-end'
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    className: styles.flexCenter,
    mr: 2
  }, /*#__PURE__*/_react["default"].createElement(_ShoppingBasket["default"], null), /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.cart
  }, "4")), /*#__PURE__*/_react["default"].createElement(_IconButton["default"], null, /*#__PURE__*/_react["default"].createElement(_ClearAll["default"], null))));
};

var _default = DailyHeader;
exports["default"] = _default;
