// @mui-treasury/mockup/brands/dailyShopping/DailyInteger.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Add = _interopRequireDefault(require("@material-ui/icons/Add"));

var _Remove = _interopRequireDefault(require("@material-ui/icons/Remove"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var useStyles = (0, _styles.makeStyles)(function (_ref) {
  var palette = _ref.palette;
  return {
    root: {
      display: 'inline-flex',
      alignItems: 'center',
      padding: 4,
      borderRadius: 40,
      border: '1px solid',
      borderColor: palette.grey[300]
    },
    iconBtn: {
      padding: 8,
      '& svg': {
        fontSize: 16
      }
    },
    value: {
      padding: '0px 8px'
    }
  };
});

var DailyInteger = function DailyInteger(_ref2) {
  var children = _ref2.children;
  var styles = useStyles();
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.root
  }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    className: styles.iconBtn
  }, /*#__PURE__*/_react["default"].createElement(_Remove["default"], null)), /*#__PURE__*/_react["default"].createElement("span", {
    className: styles.value
  }, children), /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    className: styles.iconBtn
  }, /*#__PURE__*/_react["default"].createElement(_Add["default"], null)));
};

var _default = DailyInteger;
exports["default"] = _default;
