// @mui-treasury/mockup/brands/dailyShopping/DailySearch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _InputAdornment = _interopRequireDefault(require("@material-ui/core/InputAdornment"));

var _InputBase = _interopRequireDefault(require("@material-ui/core/InputBase"));

var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var useStyles = (0, _styles.makeStyles)(function (_ref) {
  var _root;

  var spacing = _ref.spacing,
      breakpoints = _ref.breakpoints;
  return {
    root: (_root = {
      backgroundColor: '#F6F7F9',
      padding: spacing(1, 3),
      borderRadius: 40,
      flexShrink: 0,
      display: 'none'
    }, _defineProperty(_root, breakpoints.up('sm'), {
      display: 'inline-flex'
    }), _defineProperty(_root, breakpoints.up('md'), {
      minWidth: 300
    }), _root),
    iconBtn: _defineProperty({}, breakpoints.up('sm'), {
      display: 'none'
    })
  };
});

var DailySearch = function DailySearch() {
  var styles = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_InputBase["default"], {
    className: styles.root,
    placeholder: 'Search',
    endAdornment: /*#__PURE__*/_react["default"].createElement(_InputAdornment["default"], {
      position: 'end'
    }, /*#__PURE__*/_react["default"].createElement(_Search["default"], null))
  }), /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    className: styles.iconBtn
  }, /*#__PURE__*/_react["default"].createElement(_Search["default"], null)));
};

var _default = DailySearch;
exports["default"] = _default;
