// @mui-treasury/mockup/brands/dailyShopping/DailyCart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Box = _interopRequireDefault(require("@material-ui/core/Box"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _Table = _interopRequireDefault(require("@material-ui/core/Table"));

var _TableBody = _interopRequireDefault(require("@material-ui/core/TableBody"));

var _TableCell = _interopRequireDefault(require("@material-ui/core/TableCell"));

var _TableContainer = _interopRequireDefault(require("@material-ui/core/TableContainer"));

var _TableHead = _interopRequireDefault(require("@material-ui/core/TableHead"));

var _TableRow = _interopRequireDefault(require("@material-ui/core/TableRow"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));

var _DailySizeSelector = _interopRequireDefault(require("./DailySizeSelector"));

var _DailyInteger = _interopRequireDefault(require("./DailyInteger"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var useStyles = (0, _styles.makeStyles)(function (_ref) {
  var _heading;

  var palette = _ref.palette,
      breakpoints = _ref.breakpoints;
  return {
    heading: (_heading = {
      fontWeight: 900,
      fontSize: '1.75rem',
      textAlign: 'center'
    }, _defineProperty(_heading, breakpoints.up('sm'), {
      textAlign: 'left'
    }), _defineProperty(_heading, breakpoints.up('md'), {
      fontSize: '2.25rem'
    }), _heading),
    table: {
      minWidth: 650
    },
    image: {
      width: '100%',
      height: '100%',
      objectFit: 'contain'
    },
    name: {
      fontFamily: '-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif',
      fontWeight: 'bold',
      fontSize: 16,
      margin: '0 0 8px 0'
    },
    descr: {
      fontSize: 14,
      color: palette.text.secondary
    }
  };
});

function createData(image, name, descr, size, quantity, totalPrice) {
  return {
    image: image,
    name: name,
    descr: descr,
    size: size,
    quantity: quantity,
    totalPrice: totalPrice
  };
}

var rows = [createData('https://dynamic.zacdn.com/TIqU0jk90hPxnuO44NnNXO4B1AU=/fit-in/346x500/filters:quality(95):fill(ffffff)/http://static.sg.zalora.net/p/fila-4662-609589-1.jpg', 'Henry T-Shirt', 'White, Screen', 'S', 2, '$39.98'), createData('https://www.hybridoutfitters.com/wp-content/uploads/2019/11/147.jpg', 'Stripe Tee', 'ocean, stripe', 'M', 3, '$100'), createData('https://gloimg.zafcdn.com/zaful/pdm-product-pic/Clothing/2019/09/18/goods-first-img/1568766431491927776.jpg', 'Sweater Hood', 'Light Brown, Wool', 'S', 1, '$39.99'), createData('https://l.lnwfile.com/_resize_images/600/600/w1/nh/5z.jpg', 'Jackboot', 'Brown, Leather', 10.5, 1, '$69.99')];

var DailyCart = function DailyCart() {
  var styles = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    pt: {
      xs: 2,
      sm: 4,
      md: 6
    }
  }, /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    className: styles.heading,
    variant: 'h1',
    gutterBottom: true
  }, "Shopping Cart."), /*#__PURE__*/_react["default"].createElement(_TableContainer["default"], null, /*#__PURE__*/_react["default"].createElement(_Table["default"], {
    className: styles.table,
    "aria-label": "simple table"
  }, /*#__PURE__*/_react["default"].createElement(_TableHead["default"], null, /*#__PURE__*/_react["default"].createElement(_TableRow["default"], null, /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, "Product"), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, "Size"), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, "Quantity"), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, "Total Price"), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, "\xA0"))), /*#__PURE__*/_react["default"].createElement(_TableBody["default"], null, rows.map(function (row) {
    return /*#__PURE__*/_react["default"].createElement(_TableRow["default"], {
      key: row.name
    }, /*#__PURE__*/_react["default"].createElement(_TableCell["default"], {
      component: "th",
      scope: "row"
    }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
      display: 'flex',
      alignItems: 'center'
    }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
      width: 80,
      height: 80
    }, /*#__PURE__*/_react["default"].createElement("img", {
      className: styles.image,
      alt: row.name,
      src: row.image
    })), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
      ml: 2
    }, /*#__PURE__*/_react["default"].createElement("p", {
      className: styles.name
    }, row.name), /*#__PURE__*/_react["default"].createElement("span", {
      className: styles.descr
    }, row.descr)))), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, /*#__PURE__*/_react["default"].createElement(_DailySizeSelector["default"], null, row.size)), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, /*#__PURE__*/_react["default"].createElement(_DailyInteger["default"], null, row.quantity)), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, row.totalPrice), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], null, /*#__PURE__*/_react["default"].createElement(_Close["default"], null))));
  })))));
};

var _default = DailyCart;
exports["default"] = _default;
