// @mui-treasury/mockup/brands/messenger/ChatsHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Box = _interopRequireDefault(require("@material-ui/core/Box"));

var _Avatar = _interopRequireDefault(require("@material-ui/core/Avatar"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _SettingsApplications = _interopRequireDefault(require("@material-ui/icons/SettingsApplications"));

var _Edit = _interopRequireDefault(require("@material-ui/icons/Edit"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var useStyles = (0, _styles.makeStyles)(function () {
  return {
    middle: {
      flex: 'auto',
      marginLeft: 16
    },
    iconBtn: {
      padding: 8,
      backgroundColor: 'rgba(0, 0, 0, .04)',
      '&:not(:last-child)': {
        marginRight: 16
      }
    }
  };
});

var ChatsHeader = function ChatsHeader() {
  var styles = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    py: '10px',
    px: 2,
    display: 'flex',
    alignItems: 'center'
  }, /*#__PURE__*/_react["default"].createElement(_Avatar["default"], {
    alt: 'me',
    src: 'https://i.pravatar.cc/300?img=10'
  }), /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    variant: 'h5',
    className: styles.middle
  }, /*#__PURE__*/_react["default"].createElement("b", null, "Chats")), /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    className: styles.iconBtn
  }, /*#__PURE__*/_react["default"].createElement(_SettingsApplications["default"], null)), /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    className: styles.iconBtn
  }, /*#__PURE__*/_react["default"].createElement(_Edit["default"], null)));
};

var _default = ChatsHeader;
exports["default"] = _default;
