// @mui-treasury/mockup/brands/reactjs/ReactNextArticle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Container = _interopRequireDefault(require("@material-ui/core/Container"));

var _Box = _interopRequireDefault(require("@material-ui/core/Box"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _Link = _interopRequireDefault(require("@material-ui/core/Link"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ReactNextArticle = function ReactNextArticle() {
  return /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    bgcolor: 'rgb(40, 44, 52)'
  }, /*#__PURE__*/_react["default"].createElement(_Container["default"], {
    disableGutters: true
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    mr: {
      xs: 0,
      md: 10
    },
    py: 5,
    px: {
      xs: 2,
      sm: 3,
      md: 0
    },
    textAlign: 'right',
    color: 'common.white'
  }, /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    color: 'secondary',
    variant: 'subtitle2'
  }, "Next article"), /*#__PURE__*/_react["default"].createElement(_Link["default"], {
    href: 'https://reactjs.org/docs/add-react-to-a-website.html',
    target: '_blank',
    rel: 'noopener',
    color: 'inherit',
    style: {
      fontSize: 30
    }
  }, "Add React to a Website"))));
};

var _default = ReactNextArticle;
exports["default"] = _default;
