// @mui-treasury/mockup/layout/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ContentMockUp", {
  enumerable: true,
  get: function get() {
    return _ContentEx["default"];
  }
});
Object.defineProperty(exports, "ContentFormMockUp", {
  enumerable: true,
  get: function get() {
    return _ContentForm["default"];
  }
});
Object.defineProperty(exports, "FooterMockUp", {
  enumerable: true,
  get: function get() {
    return _FooterEx["default"];
  }
});
Object.defineProperty(exports, "HeaderMockUp", {
  enumerable: true,
  get: function get() {
    return _HeaderEx["default"];
  }
});
Object.defineProperty(exports, "NavContentMockUp", {
  enumerable: true,
  get: function get() {
    return _NavContentEx["default"];
  }
});
Object.defineProperty(exports, "NavHeaderMockUp", {
  enumerable: true,
  get: function get() {
    return _NavHeaderEx["default"];
  }
});

var _ContentEx = _interopRequireDefault(require("./ContentEx"));

var _ContentForm = _interopRequireDefault(require("./ContentForm"));

var _FooterEx = _interopRequireDefault(require("./FooterEx"));

var _HeaderEx = _interopRequireDefault(require("./HeaderEx"));

var _NavContentEx = _interopRequireDefault(require("./NavContentEx"));

var _NavHeaderEx = _interopRequireDefault(require("./NavHeaderEx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
